### ISSUE TITLE
Caching working incorrectly with files with the same name

### ISSUE THREAD

**Issue by splatterb0y** [RodgerLeblanc/WebImageView#1]
Hey Rodger,

I discovered a bug using your WebImageView in my Cascades App. 

If you grab images from an URL like these:
https://img.youtube.com/vi/imp0BqHBxyQ/hqdefault.jpg
https://img.youtube.com/vi/SbN9Bppmvxk/hqdefault.jpg

It will give you the first image again and again. 
Looks like something with the "cache key" determination isn't working as intended.

**Comment by RodgerLeblanc** [comment]
Hummm, I see, the code presume it will always be a different filename, if it's the same filename, it pretends it's the same old file and doesn't update it to save brandwith. But in this case it's not what we expect it to do... To fix it for your use case, in setUrl() function, just remove this line (with its ending} bracket) :if (!imageFile.exists()) {

**Comment by splatterb0y** [comment]
I figured another solution could solve the problem. What do you think about replacing the filename generation to something like this:

``` c
QString fileName = QString("%1.%2").arg(mHash->hash(url.toString().toAscii(), QCryptographicHash::Md5).toHex(), url.toString().split("/").last().split(".").last());

```

**Comment by RodgerLeblanc** [comment]
Yes, that's a good idea, I'll update it soon with the relevant change.Thanks!

**Comment by splatterb0y** [comment]
I'll BBM you later (GMT+1 here) :smile: 
Let's not spam the Issue Tracker with this kind of conversation. :laughing:
